tests/platform_tests/sfp/test_sfputil.py

```diff
@@ -43,6 +43,22 @@ def test_check_sfputil_presence(duthosts, enum_rand_one_per_hwsku_frontend_hostn
         assert intf in parsed_presence, "Interface is not in output of '{}'".format(cmd_sfp_presence)
         assert parsed_presence[intf] == "Present", "Interface presence is not 'Present'"
 
+@pytest.mark.parametrize("cmd_sfp_error_status", ["sudo sfputil show error-status", "sudo sfputil show error-status --fetch-from-hardware"])
+def test_check_sfputil_error_status(duthosts, enum_rand_one_per_hwsku_frontend_hostname, enum_frontend_asic_index, conn_graph_facts, cmd_sfp_error_status):
+    """
+    @summary: Check SFP error status using 'sfputil show error-status'
+    """
+    duthost = duthosts[enum_rand_one_per_hwsku_frontend_hostname]
+    portmap, dev_conn = get_dev_conn(duthost, conn_graph_facts, enum_frontend_asic_index)
+
+    logging.info("Check output of '{}'".format(cmd_sfp_error_status))
+    sfp_error_status = duthost.command(cmd_sfp_error_status)
+    parsed_presence = parse_output(sfp_error_status["stdout_lines"][2:])
+    for intf in dev_conn:
+        assert intf in parsed_presence, "Interface is not in output of '{}'".format(cmd_sfp_presence)
+        assert parsed_presence[intf] == "OK", "Interface error status is not 'OK'"
```

Comment:
@stephenxs is it correct that you assume all should be OK and if it is not the test should fail? Will it also fail if the API is not supported by the vendor? should we in this case will not fail the test but rather skip it?

Reply:
Skip if "Not implemented" found in the output.